### PR TITLE
css: Enable `jump-start`, `jump-both`, `jump-end`, `jump-none` step-position

### LINKED ISSUE #43060: We support `start` and `end` but not its alias `jump-start`, `jump-end`
**To Reproduce:**
```html
<!DOCTYPE html>
<html>
<head>
<style>
  .track { width: 350px; border: 1px solid #ccc; margin: 20px; position: relative; height: 30px; }
  .box { 
    width: 50px; height: 30px; background: red; position: absolute;
    animation: move 3s infinite;
  }

  .jump-start { animation-timing-function: steps(3, jump-start); }
  .jump-end   { animation-timing-function: steps(3, jump-end); }

  @keyframes move {
    from { left: 0px; }
    to   { left: 300px; }
  }
</style>
</head>
<body>

  <div class="track">
    <b>jump-start:</b>
    <div class="box jump-start"></div>
  </div>
  <p><i>Instantly jumps to 100px. You never see it at 0px.</i></p>

  <div class="track">
    <b>jump-end (default):</b>
    <div class="box jump-end"></div>
  </div>
  <p><i>Stays at 0px for 1 full second, then jumps.</i></p>

</body>
</html>
```

### PR DESCRIPTION
**Pull request by yezhizhen**
Bumps stylo to https://github.com/servo/stylo/pull/323
As tested with examples, all step-position keywords now work: https://drafts.csswg.org/css-easing-2/#typedef-step-position

Testing: New passing WPT. Several in `moveBefore` goes from TIMEOUT to `FAIL`, as the API is not enabled in Servo. Added 4 new tests to test the proper rendering.
Fixes: #43060

### CODE REVIEW

**Comment by TimvdLippe** [comment]
FYI @lukewarlow about `moveBefore` tests

**Review by xiaochengh — COMMENTED** [review]
Just let me try to understand the situation: these keywords are already implemented in Servo, and the only thing we need to do is to enable parsing?

**Comment by yezhizhen** [comment]
> the only thing we need to do is to enable parsing

@xiaochengh Indeed. They were tested with examples in #43060. 
The moment you discover that only some keyword are supported make you wonder how is that possible, 
since these are all just animation with _discrete_ easing functions. 
We even support continuous `cubic-beizer` timing function, which is more complex.

https://github.com/user-attachments/assets/5ea03cd4-330a-452c-bed6-e6c459f16e3e

https://github.com/user-attachments/assets/71745b61-9923-49df-9aba-0869a0a1135f

**Comment by yezhizhen** [comment]
> these keywords are already implemented in Servo

@xiaochengh The animation progress calculation is done In Stylo instead of Servo: [relevant stylo code](https://github.com/servo/stylo/blob/ecf17b54251b2f380a186780811f3c234308f57d/style/servo/animation.rs#L73-L75)


All servo did is retrieving the computed `animation_declarations`:
https://github.com/servo/servo/blob/42d228dcc78f40998d5c4959085ebd6296f333d1/components/script/layout_dom/element.rs#L280-L282

File: tests/wpt/meta/css/css-easing/timing-functions-syntax-computed.html.ini [hunk 1]
```diff
@@ -1,16 +1,4 @@
...
-    expected: FAIL
-
-  [Property animation-timing-function value 'steps(2, jump-both)']
-    expected: FAIL
-
-  [Property animation-timing-function value 'steps(2, jump-none)']
-    expected: FAIL
-
   [Property animation-timing-function value 'steps(calc(2 + sign(100em - 1px)), end)']
     expected: FAIL
```
Comment (Loirooriol) [inline tests/wpt/meta/css/css-easing/timing-functions-syntax-computed.html.ini:3]: Why is this one still failing?

Comment (yezhizhen) [reply tests/wpt/meta/css/css-easing/timing-functions-syntax-computed.html.ini:3]: [wpt.fyi result](https://wpt.fyi/results/css/css-easing/timing-functions-syntax-computed.html?q=timing-functions-syntax-computed.html&run_id=5132946534301696&run_id=5125701008359424&run_id=4841128588476416&run_id=6241884055011328)

Firefox also fail these. I suspect something wrong with `calc`.

Comment (yezhizhen) [reply tests/wpt/meta/css/css-easing/timing-functions-syntax-computed.html.ini:3]: It seems this is a known bug:
https://bugzilla.mozilla.org/show_bug.cgi?id=1915976

**Review by Loirooriol — APPROVED** [review]

**Comment by yezhizhen** [comment]
@Loirooriol Do you think we should add 4 ref tests? My idea is to use comibination of negative delay (necessary to distinguish between jump-none and jump-end) to test certain frame.

Right now these tests are written with Web Animations API, which we don't support... None of the passing tests involve actual animation rendering.
https://github.com/servo/servo/blob/9394cdfa4c204349843a11b20db0fce574d78c5a/tests/wpt/tests/css/css-easing/step-timing-functions-output.html#L106-L118

**Comment by Loirooriol** [comment]
Sounds good. Indeed using paused animations with a negative delay, you can check the outcome at the desired %

**Comment by yezhizhen** [comment]
> Sounds good. Indeed using paused animations with a negative delay, you can check the outcome at the desired %

Speaking of which, we have another bug: https://github.com/servo/servo/issues/43080, that blocks us from testing multiple % in one test.. Guess we just test one single critical then..

**Comment by Loirooriol** [comment]
>  that blocks us from testing multiple % in one test

You can still do something like https://wpt.live/css/css-variables/variable-animation-guaranteed-invalid.html

**Comment by yezhizhen** [comment]
Nice. I can use our newly implemented `@property` instead, to allow the interpolation of the `--width` variable :)

**Comment by yezhizhen** [comment]
The failing subtests is because of #43086. This is ready for review.

File: tests/wpt/tests/css/css-easing/step-jump-both-servo.html [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    width: var(--width);
+    height: 20px;
+    background: green;
+    margin-bottom: 5px;
+    animation: step-anim 3s steps(3, jump-both) paused;
+    animation-delay: calc(var(--at) * -1s);
+  }
+</style>
+
+<div class="test" title="t=0: initial jump" style="--at: 0" data-expected-width="25"></div>
```
Comment (Loirooriol) [inline tests/wpt/tests/css/css-easing/step-jump-both-servo.html]: You aren't using a title anywhere else and it would be hard to notice anyways, so I would just remove it

```suggestion
<div class="test" style="--at: 0" data-expected-width="25"></div>
```

File: tests/wpt/tests/css/css-easing/step-jump-both-servo.html [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+<div class="test" style="--at: 0.5" data-expected-width="25"></div>
```
Comment (Loirooriol) [inline tests/wpt/tests/css/css-easing/step-jump-both-servo.html]: Better 0.9 to cover a bigger range

```suggestion
<div class="test" style="--at: 0.9" data-expected-width="25"></div>
```

File: tests/wpt/tests/css/css-easing/step-jump-both-servo.html [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+<div class="test" style="--at: 1" data-expected-width="50"></div>
+<div class="test" style="--at: 1.9" data-expected-width="50"></div>
+
+
+<div class="test" style="--at: 2" data-expected-width="75"></div>
```
Comment (Loirooriol) [inline tests/wpt/tests/css/css-easing/step-jump-both-servo.html]:
```suggestion
<div class="test" style="--at: 2.0" data-expected-width="75"></div>
<div class="test" style="--at: 2.9" data-expected-width="75"></div>
```

File: tests/wpt/tests/css/css-easing/step-jump-both-servo.html [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+<div class="test" style="--at: 3.0" data-expected-width="10"></div>
+
+
+<script>
+  checkLayout(".test");
+</script>
```
Comment (Loirooriol) [inline tests/wpt/tests/css/css-easing/step-jump-both-servo.html]: Add newline at the end

File: tests/wpt/tests/css/css-easing/step-jump-both-servo.html [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/check-layout-th.js"></script>
+
+<style>
+  @property --width {
+    syntax: '<length>';
+    inherits: false;
+    initial-value: 10px;
+  }
```
Comment (Loirooriol) [inline tests/wpt/tests/css/css-easing/step-jump-both-servo.html]: I would remove this, animate `width` instead

```css
  @keyframes step-anim {
    0% {
      width: 0px;
    }
    100% {
      width: 100px;
    }
  }
```

and for the initial value just set it to `.test`

```css
.test {
    width: 10px;
}
```

File: tests/wpt/tests/css/css-easing/step-jump-both.html
Comment (Loirooriol) [inline tests/wpt/tests/css/css-easing/step-jump-both.html:1]: Better don't use "servo" in the filename. It should be about what's being tested, not which engine the test is coming from.

Comment (yezhizhen) [reply tests/wpt/tests/css/css-easing/step-jump-both.html:1]: Cool. I see tests with "chrome" in the name in this folder too

**Review by xiaochengh — APPROVED** [review]

File: tests/wpt/tests/css/css-easing/step-jump-start.html [hunk 7]
```diff
@@ -0,0 +1,44 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<title>CSS Animations Test: animation-timing-function steps(3, jump-start)</title>
+<link rel="author" title="Euclid Ye" href="mailto:yezhizhenjiakang@gmail.com">
+<link rel="help" href="https://github.com">
+<link rel="help" href="https://drafts.csswg.org">
```
Comment (Loirooriol) [inline tests/wpt/tests/css/css-easing/step-jump-start.html]: Incomplete links?

**Review by Loirooriol — APPROVED** [review]

**Comment by yezhizhen** [comment]
Merging after upstream task-cluster finishes.

EDIT: DCO never started..
